.github/workflows/ci.yml

```diff
@@ -0,0 +1,92 @@
+name: Run checks on pull requests and pushes on main branch
+
+on:
+  pull_request: {}
+  push:
+    branches: [main]
```

Comment:
currently any pull request and any PR merges to `main` will trigger this workflow. If we will need new steps when merging to main e.g publish version. we can add new yml file and separate them in the future